kahuna/public/js/directives/gr-file-change.js

```diff
@@ -0,0 +1,15 @@
+import angular from 'angular';
+
+export var fileChange = angular.module('gr.fileChange', []);
+
+fileChange.directive('grFileChange', ['$parse', function($parse) {
+    return {
+        restrict: 'A',
+        link: function(scope, element, attrs) {
+            element.on('change', function() {
+                const onchange = $parse(attrs.grFileChange)(scope);
+                onchange(Array.from(element[0].files));
```

Comment:
I prefer this - we've done it else where. I wrote this when I first did Angular - apologies for the gumpf.

It makes it more explicit in the view, thus controller, what you should be expecting.

Changed.